### PR TITLE
Devlopment Coding Docs initial start.

### PR DESCRIPTION
**Pull request by jessbreckenridge**
## Overview

After cloning the repository I was struggling with how to get started working with the `gh-ost` codebase. 

I have submitted this PR in order to start a page guiding prospective contributers on how to get started with `gh-ost` development.

While this page is simple for now, I am hoping that it is a good springboard to further PR's on this documentation.

Related issue: https://github.com/github/gh-ost/issues/412

### Description

This PR creates a `doc/coding-ghost.md` file that contains instructions for running `cibuild`. It also details the steps that `cibuild` performs.

### CODE REVIEW

File: doc/coding-ghost.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+- This will bootstrap the environment if needed, format the code, build the code, and then run the unit test.
+
+## CI build workflow
+
+`script/cibuild` performs the following actions:
+
+- It runs `script/bootstrap`
+- `script/bootstrap` runs `script/ensure-go-installed`
+- `script/ensure-go-installed` installs go locally if (go is not installed) || (go is not version 1.7). It will not install go if it is already installed locally and at the correct version.
+- `script/build` builds the `gh-ost` binary and places in in `bin/`
```
Comment (shlomi-noach) [inline doc/coding-ghost.md]: I think enumerating the exact steps taken by the script is error prone. e.g. what happens if the script ensures go `1.8` -- will this text be updated?

File: doc/coding-ghost.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Getting started with gh-ost development.
+
+## Overview
+
+Getting started with gh-ost development is simple!
+
+- First clone the repository.
```
Comment (shlomi-noach) [inline doc/coding-ghost.md]: Should we clone via `git clone` or via `go get`?

File: README.md [hunk 3]
```diff
@@ -84,6 +84,8 @@ But then a rare genetic mutation happened, and the `c` transformed into `t`. And
 
 We develop `gh-ost` at GitHub and for the community. We may have different priorities than others. From time to time we may suggest a contribution that is not on our immediate roadmap but which may appeal to others.
 
+Please see [Coding gh-ost](https://github.com/github/gh-ost/blob/develdocs/doc/command-line-flags.md) for a guide to getting started developing with gh-ost.
```
Comment (shlomi-noach) [inline README.md]: :+1:

File: doc/coding-ghost.md (hunk 1 above)
Comment (jessbreckenridge) [inline doc/coding-ghost.md]: Good point, let me strip this down.

**Comment by jessbreckenridge** [comment]
@shlomi-noach Are additional changes needed before merging this PR?

**Pull request merged**